tests/test_fork.py

```diff
@@ -0,0 +1,41 @@
+# SPDX-License-Identifier: (GPL-2.0 OR Linux-OpenIB)
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All rights reserved.
+
+import errno
+import pyverbs.enums as e
+from pyverbs.fork import fork_init, is_fork_initialized
+from pyverbs.pyverbs_error import PyverbsRDMAError
+
+from tests.base import PyverbsAPITestCase
+
+
+class ForkAPITest(PyverbsAPITestCase):
+    """
+    Test the API of the fork functions.
+    """
+
+    def setUp(self):
+        super().setUp()
+
+    def test_fork_init(self):
+        try:
+            fork_init()
+        except PyverbsRDMAError as ex:
+            # Depends on the order of the tests EINVAL could be returned if
+            # fork_init() is called after an MR has already been registered.
+            self.assertEqual(ex.error_code, errno.EINVAL)
+
+    def test_is_fork_initialized(self):
+        try:
+            fork_init()
+            expected_ret = [e.IBV_FORK_ENABLED, e.IBV_FORK_UNNEEDED]
```

Comment:
you can call is_fork_initialized() before calling fork_init and check its returned value to make a more "precise" check after fork_init.

Reply:
I didn't want to get into corner cases with ordering of the different test cases.
If a prior test case already called fork_init() then is_fork_initialized() could return IBV_FORK_ENABLED, even if fork_init() is not called yet in this test case.

Reply:
I know, what I meant, is to call is_fork_initialized(), according to the returned value you would know the exact expected value of the second is_fork_initialized call after fork_init().
But yes, then you're required to handle multiple cases.
You can keep it as is... it wouldn't have much added value.